orchestrator/domain/base.py

```diff
@@ -865,6 +930,29 @@ def from_product_id(
             **instances,  # type: ignore
         )
 
+    @classmethod
+    def from_other_lifecycle(
+        cls: Type[S],
+        other: "SubscriptionModel",
+        status: SubscriptionLifecycle,
+    ) -> S:
+        if not cls.__base_type__:
+            # Import here to prevent cyclic imports
+            from orchestrator.domain import SUBSCRIPTION_MODEL_REGISTRY
+
+            cls = SUBSCRIPTION_MODEL_REGISTRY.get(other.product.name, cls)  # type:ignore
+            cls = lookup_specialized_type(cls, status)
+
+        data = cls._data_from_lifecycle(other, status)
+
+        data["status"] = status
+        if data["start_date"] is None and status == SubscriptionLifecycle.ACTIVE:
+            data["start_date"] = nowtz()
```

Comment:
What's the use case for an active sub with no start date?

Reply:
There is none. This code sets the start date if not set and the sub becomes active

Reply:
Let me rephrase: how can a active sub exists without start_date: or is this code just existing to handle that race condition

Reply:
it cannot  (unless you manipulate the db yourself. But this code also runs for provisioning -> active and provisioning can lack a start date